File: 942-di-string-match/942-di-string-match.rb
# @param {String} s
# @return {Integer[]}
def di_string_match(s)
    res = []
    low, high = 0, s.length
    s.each_char do |char|
       if char == 'I'
           res << low
           low += 1
       else
           res << high
           high -= 1
       end
    end
    res << low
end

File: 942-di-string-match/942-di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        vector<int> res;
        int low = 0, high = s.size(), i = 0;
        while (low <= high)
        {
            if (s[i] == 'I')
            {
                res.push_back(low);
                low++;
            }
            else{
                res.push_back(high);
                high--;
            }
            i++;
        }
        return res;
    }
};